### PR TITLE
UIUX 64: Local storage for Theme

### PR DESCRIPTION
**Pull request by ariadev79**

### CODE REVIEW

File: frontend_new/src/App.tsx [hunk 1]
```diff
@@ -9,18 +9,22 @@ import getDefaultTheme from "./components/ThemeSetting";
 
 //using colors from theme - bit hacky but works
 function App() {
+  let themeColourSaved = localStorage.getItem("themeChoice");
+  let themeColor =
+    themeColourSaved == null ? getDefaultTheme() : themeColourSaved;
```
Comment (EngineersBox) [inline frontend_new/src/App.tsx]: **[Desired]** Can be simplified to:
```javascript
let themeColor = localStorage.getItem("themeChoice") || getDefaultTheme();
```

**Review by EngineersBox — COMMENTED** [review]
A minor simplification of the theme retrieval. Otherwise looks good. I won't approve in favour of Jay reviewing it instead.

File: frontend_new/src/components/Cards.tsx [hunk 2]
```diff
@@ -28,20 +28,51 @@ function gotoLink(url: string) {
...
+  // 2. the articles array with json article data to map  during bookmarks
   function handlebookmark() {
-    props.handlebookmark(
-      props.web_link,
-      props.primary,
-      props.secondary,
-      props.id
-    );
-    setBookmarked(props.id in props.bookmarks);
+    if (localStorage.getItem(props.id.toString()) == null) {
```
Comment (jayjayjayjayjayjay) [inline frontend_new/src/components/Cards.tsx]: might be good to split out all this code into a seperate file and import into Cards.tsx. If you have no time to do it, I can just merge it

Comment (ariadev79) [reply frontend_new/src/components/Cards.tsx]: Yea sort that out later. For now it's still bearable

File: frontend_new/src/components/BookmarkDrawer.tsx [hunk 3]
```diff
@@ -96,6 +96,20 @@ export default function PersistentDrawerLeft(props: props) {
...
+  );
+
+  // To handle the case where the localStorage is empty {}
+  bookmarkedArticles =
+    bookmarkedArticles != null && Object.keys(bookmarkedArticles).length > 0
+      ? bookmarkedArticles
+      : null;
+
+  // Need to declare any as the Object.values(bookmarkedArticles) returns unknown type
+  let valuesOfBookmarkedArticles: any =
```
Comment (jayjayjayjayjayjay) [inline frontend_new/src/components/BookmarkDrawer.tsx]: what type is valueofBookmarkedArticles? Preferable to not write any

Comment (ariadev79) [reply frontend_new/src/components/BookmarkDrawer.tsx]: Removed this variable

**Pull request merged**